### PR TITLE
chore: migrate to v1 APIs

### PR DESCRIPTION
**Pull request by jmdeal**
Fixes #N/A 

**Description**
This PR updates all in-code references to the `v1beta1` APIs to `v1`. This is largely just find and replace, but there are a few changes to note:
- `APIVersion` is updated to `Group` in `NodeClassRef`
- `KubeletConfig` is first parsed from the compat annotation, and then the new field on the `EC2NodeClass`
- Changed the storage version from `v1beta1` to `v1` for CRDs

We've also updated the naming for package aliases due to the new `v1` conflicts. These packages have new names:

|package|old alias|new alias|
|-|-|-|
|k8s.io/api/core/v1|v1|corev1|
|github.com/aws/karpenter-provider-aws/pkg/apis/v1(beta1)|v1beta1|v1(beta1)|
|sigs.k8s.io/karpenter/pkg/apis/v1(beta1)|corev1beta1|karpv1(beta1)|

**How was this change tested?**
`make presubmit` and `/karpenter snapshot`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

File: pkg/providers/launchtemplate/suite_test.go [hunk 1]
```diff
@@ -122,14 +122,17 @@ var _ = AfterEach(func() {
...
+	var nodePool *karpv1.NodePool
+	var nodeClass *v1.EC2NodeClass
 	BeforeEach(func() {
 		nodeClass = test.EC2NodeClass(
-			v1beta1.EC2NodeClass{
-				Status: v1beta1.EC2NodeClassStatus{
+			v1.EC2NodeClass{
+				// Spec: v1.EC2NodeClassSpec{
+				// 	Kubelet: &v1.KubeletConfiguration{},
+				// },
```
Comment (njtran) [inline pkg/providers/launchtemplate/suite_test.go:132]: remove

**Review by njtran — COMMENTED** [review]
Really nice work! Thanks for doing this. I think I found some artifacts of your testing

File: pkg/controllers/interruption/interruption_benchmark_test.go [hunk 2]
```diff
@@ -302,16 +302,16 @@ func makeStateChangeMessagesAndNodes(count int, states []string) ([]interface{},
...
+func makeSpotInterruptionMessagesAndNodes(count int) ([]interface{}, []*corev1.Node) {
 	var msgs []interface{}
-	var nodes []*v1.Node
+	var nodes []*corev1.Node
 	for i := 0; i < count; i++ {
 		instanceID := fake.InstanceID()
 		msgs = append(msgs, spotInterruptionMessage(instanceID))
 		nodes = append(nodes, coretest.Node(coretest.NodeOptions{
-			ObjectMeta: metav1.ObjectMeta{
+			ObjectMeta: metacorev1.ObjectMeta{
```
[the remark this answers is not in the document]
Comment (jmdeal) [reply pkg/controllers/interruption/interruption_benchmark_test.go]: fixed

File: pkg/providers/amifamily/resolver.go [hunk 3]
```diff
@@ -205,25 +205,27 @@ func (r Resolver) defaultClusterDNS(opts *Options, kubeletConfig *corev1beta1.Ku
...
-			return nil, err
-		}
+	kubeletConfig, err := utils.GetKubeletConfigurationWithNodeClaim(nodeClaim, nodeClass)
+	if err != nil {
+		return nil, fmt.Errorf("resolving kubelet configuration, %w", err)
+	}
+	if kubeletConfig == nil {
+		kubeletConfig = &v1.KubeletConfiguration{}
 	}
+	fmt.Printf("resolved kubelet config with maxPods %d\n", lo.FromPtr(kubeletConfig.MaxPods))
```
[the remark this answers is not in the document]
Comment (jmdeal) [reply pkg/providers/amifamily/resolver.go:217]: Nope, will remove

File: pkg/controllers/interruption/interruption_benchmark_test.go [hunk 4]
```diff
@@ -283,17 +283,17 @@ func makeScheduledChangeMessagesAndNodes(count int) ([]interface{}, []*v1.Node)
...
 	var msgs []interface{}
-	var nodes []*v1.Node
+	var nodes []*corev1.Node
 	for i := 0; i < count; i++ {
 		state := states[r.Intn(len(states))]
 		instanceID := fake.InstanceID()
 		msgs = append(msgs, stateChangeMessage(instanceID, state))
 		nodes = append(nodes, coretest.Node(coretest.NodeOptions{
-			ObjectMeta: metav1.ObjectMeta{
+			ObjectMeta: metacorev1.ObjectMeta{
```
[the remark this answers is not in the document]
Comment (jmdeal) [reply pkg/controllers/interruption/interruption_benchmark_test.go]: fixed

**Review by njtran — APPROVED** [review]

**Review by njtran — APPROVED** [review]
LGTM

**Pull request merged**